### PR TITLE
feat: add support for extension to sign order payload

### LINKED ISSUE #1227: Realtime Event for Order with funding address
## 📝 Bug Description

In history page, when user cancels the order from other trading account, order gets cancelled but there is no realtime event so that it can be removed from UI immediately. 
We will be allowing subscription for selected `mainAddress` so that Order events could work properly in history page.

### LINKED ISSUE #1242: P2: Wallet Connect Interaction - Extension Proxy UI changes
## 📝 Feature Description

We will be supporting trading with simply an extension account in future releases, which will significantly streamline the onboarding process for new users. To implement this functionality, certain frontend adjustments are needed. The following is a short list of the things that would be impacted by these changes: 

1. No changes required in balance page

3. In history page, for open orders tab, need to fix Cancel order functionality 
    1. User have a funding account
        1. If user have a trading account as well, then cancel based on trading account (if tradingAddress !== mainAddress)
        2. else cancel based on funding account
    2. User didn’t have funding account
        1. He won’t see any data, as all data on history page is funding account centric (Nothing to do here)

4. In exchange page, 
    1. For place order component
        1. By default, Show `Connect Trading Account` or may be rename as `Connect Wallet` button as user can now select funding account to trade
        3. User selects a funding account
            1. Mark proxy (which has same as funding account) as `Selected` by default if exists
            2. Howver, user can change the proxy account and choose a keyring from profile section
        4. If User chose a trading account directly (keyring), nothing to do
    2. For open orders, order history & trade history components, 
        1. By default, Show `Connect Trading Account` or may be rename as `Connect Wallet` button as user can now select funding account to trade
        2. If User selects a funding account
            1. Mark proxy (which has same as funding account) as `Selected` by default if exists
            2. Howver, user can change the proxy account and choose a keyring from profile section
        3. If User chose a trading account directly (keyring), fetch data for the selected keyring address

5. For transfer page,
    1. For funding to trading transfer, Trading account card shouldn’t say `Account not present` if using extension proxy

        funding to funding

    3. For Trading to funding transfer, if using extension proxy
        1. It shouldn’t say `Account not present` 
        2. It shouldn’t display `Connect your account` button
        &nbsp;
        
        trading to funding

6. In Header component, Remove the card for trading account instead, add a `Options` button.
  image

&nbsp;
7. Fix onboarding process modals. There are two cases basically - 
   - When user already have some PDEX
   - When user don't have any PDEX

Notion - https://www.notion.so/polkadex/Extension-Order-Support-388934c02ec14822aefef80a33d6fe55

### LINKED ISSUE #1246: P2.1: Connect Wallet Changes
## 📝 Feedback Description

As we have added the functionality to start trading with having only extension account, but for now user couldn't register it's funding account for trading. So, basically the connect wallet flow going to be tweeked a bit such that creating trading account is not a priority. User can just deposit the amount and start trading.

### - New User (The funding account which doesn't have any proxy registered yet)

1. Change this text and it's description & add a `Transfer` button (transfer from funding to trading which registers proxy behind the scenes in backend) if user has required PDEX balance else show `Continue` button.

    image

2. On clicking `Continue` on previous popup. No changes required here.

    image

### - Existing User (The funding account which have atleast one proxy registered)

1. No trading account found in browser 
    - Textual changes here
    - Add a 4th button saying `Register funding account` so that existing user can register thier main account for proxy & add a `NEW` label 
        &nbsp;
        image

4. Trading account found in browser
    - Add a 4th button saying `Register funding account` (same as above)
       &nbsp;
       image

## 🌍 Your Environment

| Software         | Version(s)                  |
| :----------------: | :---------------------------: |
| Project/Package  |             Orderbook Frontend                |
| Browser          | [x] Chrome [] Firefox [] Safari |
| Operating System | [] Windows [x] MacOS [] Linux   |

### LINKED ISSUE #1258: There is an issue in the orderbook page even after the user transfer the funds to trading account and navigates to orderbook page
# 📝 Bug Description
There is an issue in the orderbook page even after the user transfer the funds to the trading account and navigates to orderbook page
Verify the Sell and Buy buttons

## 🤔 Expected Behavior
If the user transfer the amount to the trading account, the the user should able to the trade

## 😯 Current Behavior
In orderbook Sell and Buy buttons displayed "Fund Account" text and not able to make a trade

image

## 🔎 Steps to Reproduce

steps
Precondition -> Create a new funding account with a balance of 2PDEX
1. Navigate to Polkadex home page
2. Select polkadot extension
3. Select a newly created funding account
4. Displayed "Welcome to polkadex orderbook" pop-up window
5. Click the "Transfer funds" button
6. Navigate to the "Transfer" Page
7. Select the Asset 
8. Give the Amount ("0.9 PDEX") to transfer
9. Click the Transfer button
10. click the "sign and submit" button
11. Give a password for Polkadot to do the transaction
12. Click the "sign the transaction" button
13. Click Trade Tab
14. Verify the user is able to do the trade. Verify the sell and buy buttons

## 💁 Possible Solution
If the transaction does not happen or got any error then the fees should not be deducted from the funding account

## 🌍 Your Environment

| Software         | Version(s)                  |
| ---------------- | --------------------------- |
| Project/Package  |                             |
| Browser          | []Chrome []Firefox []Safari |
| Operating System | []Windows []MacOS []Linux   |

### PR DESCRIPTION
**Pull request by nuel77**
## Description - 

As of now, Orderbook needs a trading account to start trading but now we are making such changes so that creating trading account is not a must-have thing. Orderbook now supports trading with extension accounts only. User can now just deposit tokens and start trading. Basically, Creating a trading account is an additional feature now.
This PR aims to make such changes in UI so that it could be feasible for that. There are several changes done - 

- [x] New Connect Wallet flow
- [x] Profile section changes
- [x] Place/Cancel Order, Withdrawal and Cancel All hooks compatibility
- [x] Bug fixes for removing trading account
- [x] Refactor Accounts subscription
- [x] Place Order component
- [x] Fix process of Cancel Order and Cancel All Orders in History page
- [x] Fix Responsiveness as some components changed

### CODE REVIEW

File: apps/hestia/src/components/trading/Orders/index.tsx [hunk 1]
```diff
@@ -47,6 +47,12 @@ export const Orders = () => {
   } = useProfile();
   const connected = tradeAddress?.length > 0;
 
+  const isFundingButton = useMemo(() => {
+    if (!mainAddress) return true;
+    if (!tradeAddress) return false;
+    return true;
+  }, [mainAddress, tradeAddress]);
```
Comment (serhii-temchenko) [inline apps/hestia/src/components/trading/Orders/index.tsx]: Is there any need to check `if (!mainAddress) return true;` first?

Can it be simplified to:
```ts
const isFundingButton = useMemo(() => {
    return !!tradeAddress;
}, [tradeAddress]);
```

File: apps/hestia/src/components/trading/PlaceOrder/connectAccount.tsx [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    onToogleConnectExtension,
+    onToogleFundWallet,
+  } = useSettingsProvider();
+  const { mainProxiesAccounts } = useConnectWalletProvider();
+
+  const isFundingButton = useMemo(() => {
+    if (!mainAddress) return true;
+    if (!tradeAddress) return false;
+    return true;
+  }, [mainAddress, tradeAddress]);
```
Comment (serhii-temchenko) [inline apps/hestia/src/components/trading/PlaceOrder/connectAccount.tsx]: Same here

File: apps/hestia/src/components/trading/Orders/index.tsx (hunk 1 above)
Comment (ap211unitech) [reply apps/hestia/src/components/trading/Orders/index.tsx]: Well....Yes actually. One dependency could be removed here. I rather choose to removed `tradeAddress` dependency here. 

```js
const isFundingType = useMemo(() => !mainAddress, [mainAddress]);
```

File: apps/hestia/src/components/trading/PlaceOrder/connectAccount.tsx (hunk 2 above)
Comment (ap211unitech) [reply apps/hestia/src/components/trading/PlaceOrder/connectAccount.tsx]: Same

**Review by serhii-temchenko — APPROVED** [review]

**Review by minikas — CHANGES_REQUESTED** [review]
When clicking on 'import,' the component to register disappears

https://github.com/Polkadex-Substrate/Polkadex-Orderbook-Frontend/assets/12574469/67fa3277-2e90-4963-9be2-e762664bc781

**Review by minikas — COMMENTED** [review]
The 'Learn more' button redirects to the previous version of Orderbook; perhaps we should remove it or try to update the docs. @nuel77 


https://github.com/Polkadex-Substrate/Polkadex-Orderbook-Frontend/assets/12574469/671354e6-127d-4e97-a3c8-a9ceb3073d81

**Review by minikas — CHANGES_REQUESTED** [review]
This tag looks a bit odd; try reducing the text a bit more and adding some padding.

Screenshot 2024-05-15 at 07 41 00

**Comment by nuel77** [comment]
> The 'Learn more' button redirects to the previous version of Orderbook; perhaps we should remove it or try to update the docs. @nuel77
> 
>  Screen.Recording.2024-05-15.at.07.38.49.mov

Sounds good to me.

**Comment by ap211unitech** [comment]
> > The 'Learn more' button redirects to the previous version of Orderbook; perhaps we should remove it or try to update the docs. @nuel77
> > Screen.Recording.2024-05-15.at.07.38.49.mov
> 
> Sounds good to me.

What's todo here ? Should we remove it ? IMO, we should update the docs rather removing it. `Read More` button is helpful for new users..

**Comment by ap211unitech** [comment]
> When clicking on 'import,' the component to register disappears
> 
>  Screen.Recording.2024-05-15.at.07.36.07.mov

This is fixed now.

**Comment by ap211unitech** [comment]
> This tag looks a bit odd; try reducing the text a bit more and adding some padding.
> 
> Screenshot 2024-05-15 at 07 41 00

@minikas Fixed now.

**Comment by Hildajose** [comment]
@nuel77 @pavankanteti 
https://docs.google.com/spreadsheets/d/18CAPJ0djQp4syhU3-sUAvtX56o6IYWUArzZ8c5L3L34/edit?usp=sharing

attached the testcase report which I am targeting to test against the feature. please do let me know if any modification required in this.
Manual execution is in progress and I am targeting to complete by this Friday(17th May). have observed a few issues/discrepancies while making the execution and created bugs for the same.

**Review by minikas — APPROVED** [review]
LGTM

**Pull request merged**